### ISSUE TITLE
useLocalStorage will set String(undefined) if there is no defaultValue

### ISSUE THREAD

**Issue by saintwinkle** [childrentime/reactuse#88]
## Reproduction

Only trigger when there's no data in localStorage.

```jsx
const [text, setText] = useLocalStorage('text')

useEffect(() => {
  console.log(`text: ${text}`)
}, [text])

useEffect(() => {
  setTimeout(() => {
    setText('hello')
  }, 1000)
}, [setText])
```

Code above will log:

```js
// text: null
// text: undefined
         ^ 🚨 
// text: hello
```

## Reference

Probably there's something wrong with the code here:

https://github.com/childrentime/reactuse/blob/33712231b190c05f5bc913c66a946be2a6aaf322/packages/core/src/createStorage/index.ts#L142

**Comment by childrentime** [comment]
You are right. we should always set `null` for empty data,  I will change it soon

**Comment by childrentime** [comment]
fixed in v5.0.14
